base_layer/core/src/chain_storage/postgres_db/migrations/2020-04-07-13-32_block_headers/up.sql

```diff
@@ -0,0 +1,20 @@
+create table block_headers (
+    hash TEXT NOT NULL PRIMARY KEY,
```

Comment:
I originally made this a hex string so that it could queried by a partial hash. That was so that I could use it to build a block explorer. I'm not sure that that is necessary anymore and if I did it again I'd add another id field and store this as bytea. That's just my opinion

Reply:
I think the long term goal is still to be able to use this for a block explorer. But I like the idea of making the hash human readable inside of the table

Reply:
I would like to ask why use TEXT as a primary key? 

I think this will result in a lot of performance issues, particularly around re-indexing.

I would think a numerical non-clustered primary key and a clustered key for `hash` would be a better approach.

Reply:
Can you expand why that would be better?

I am thinking its wont be an issue because:
1. Its write once, rarely modified (you will only change the orphan flag) while read many. So it wont have to reindex a lot. 
2.   We cant cluster hash, as its random. If we want to cluster around an index, we might want to cluster around height. But height is not unique.

Reply:
Fair points.

I also seem to be mixing up specifics to MSSql and mySQL, with postgre.